src/function/gds/strongly_connected_components.cpp

```diff
@@ -0,0 +1,346 @@
+#include "binder/binder.h"
+#include "binder/expression/expression_util.h"
+#include "common/exception/runtime.h"
+#include "common/types/types.h"
+#include "function/gds/auxiliary_state/gds_auxilary_state.h"
+#include "function/gds/gds_frontier.h"
+#include "function/gds/gds_function_collection.h"
+#include "function/gds/gds_object_manager.h"
+#include "function/gds/gds_utils.h"
+#include "function/gds/output_writer.h"
+#include "function/gds_function.h"
+#include "gds_vertex_compute.h"
+#include "graph/graph.h"
+#include "processor/execution_context.h"
+#include "processor/result/factorized_table.h"
+
+using namespace kuzu::binder;
+using namespace kuzu::common;
+using namespace kuzu::processor;
+using namespace kuzu::storage;
+using namespace kuzu::graph;
+
+namespace kuzu {
+namespace function {
+
+static constexpr offset_t SCC_UNSET = std::numeric_limits<offset_t>::max();
+
+struct SCCComputationState {
```

Comment:
GDSAuxiliaryState is to hold algorithm specific state. So u might want to make this a subclass of GDSAuxiliaryState. Later when u wanna extend SCC to multi-tabled case, you will find this useful.

Reply:
Do you mean just making it a subclass or use it differently too?